### ISSUE TITLE
Implement `CONTINUE` convention

### ISSUE THREAD

**Issue by kadrlica** [esheldon/fitsio#118]
This follows from the discussion in #117. The `cfitsio` library has functions implemented for reading, writing and manipulating long strings following the `CONTINUE` convention described here:

https://fits.gsfc.nasa.gov/registry/continue_keyword.html

Specifically, we would want to interface with the `cfitsio` functions like 
[`fits_read_key_longstr`](https://github.com/kadrlica/fitsio/blob/master/cfitsio3370patch/longnam.h#L138)
[`fits_write_key_longstr`](https://github.com/kadrlica/fitsio/blob/master/cfitsio3370patch/longnam.h#L82)
to manipulate long sting headers.

Since this could be a significant change to how header cards are parsed (and a non-zero investment of time), I have created a new issue to track this development.

**Comment by kadrlica** [comment]
It appears that LSST is also using the `CONTINUE` convention. This probably motivates us interested parties (@kadrlica @erykoff @rmjarvis) to do a proper implementation sometime soon.

**Comment by buchbend** [comment]
For our in-house created fits files for the upGREAT instrument on SOFIA we are also using the CONTINUE convention and would be very interested in an implementation in ´fitsio´. We would greatly benefit from the increased performance over ´astropy.io.fits´, which we are currently using to parse the fits-headers.  Unfortunately, I did not have a great idea either on how to implement this myself easily, when checking the source code of fitsio.

**Comment by rmjarvis** [comment]
A start would be to submit a test that currently fails.  It's not clear to me what kind of `CONTINUE` usage is currently causing anyone problems.  I haven't run into any trouble with LSST images yet.  Without a clear description of what isn't working, it's hard to know what kind of change is required.

**Comment by buchbend** [comment]
Sorry for the vague information from my side. I am using Ubuntu 18.04 with python 3.6.5. I have two issues with the current state:

### Keywords with CONTINUE are not read as long strings

I looked at the fitsio test suite and found an existing test related to CONTINUE. So far this test only checks that the CONTINUE keys are seen. A test for an implementation of the CONTINUE convention would be to check that the value of the header keyword followed by CONTINUEs is actually the long string that is encoded in the subsequent entries. For the existing Test ` testHeaderContinue` (current master branch `38df6f0`) the following snippet added in line `371` would test a sucessfull read of the CONTINUE convention:

~~~ .py
with fitsio.FITS(fname) as fits:
    rh = fits[0].read_header()
    assert rh["SVALUE"] == "This is a long string value extending over 3 lines."
~~~

### Some versions of fitsio report `ValueError: CONTINUE not supported`

Using the fitsio version currently on pip ('0.9.11') reports `ValueError: CONTINUE not supported` when executing the following snippet (copy paste from `testHeaderContinue`):

~~~ .py
import fitsio
import tempfile
import numpy

fname=tempfile.mktemp(prefix='fitsio-HeaderContinue-',suffix='.fits')
with fitsio.FITS(fname,'rw',clobber=True) as fits:
    data=numpy.zeros(10)
    header = [                          
                "SVALUE  = 'This is a long string value &' ",
                "CONTINUE  'extending&   '   ",
                "CONTINUE  ' over 3 lines.'     / and a comment ",
                "TEST    = 10 / another key",
              ]
    fits.write_image(data, header=header)
~~~

while a checkout and install of the current master version: '1.0.0rc1' (`38df6f0`) works. However it does not read the long string as described above.

### Wrap up

The current master version reads headers with CONTINUE, however the strings are cut off. The current version on pip (at least on my machine) raises an Exception when CONTINUEs are used on the fits-header.

Sorry for putting all of this into this comment. I am happy to extend the Tests but I am not sure about a proper place. If I can be of help with the implementation of the `CONTINUE` convention I would be happy to help.

**Comment by kadrlica** [comment]
I fully agree that `fitsio` should respect the `CONTINUE` convention. I think that there was a proposed implementation on [#117](https://github.com/esheldon/fitsio/issues/117#issuecomment-297808111), but we never got around to making it a reality. What we did do was follow the recommendation of the `CONTINUE` convention documentation:
"FITS readers that do not support this convention should interpret the CONTINUE keywords as being equivalent to a COMMENT keyword. "

We should keep in mind that there is a rather large set of DES data that were processed with an older version of `fitsio` and now contain corrupted `CONTINUE` cards. These data will likely be un-readable by future versions of `fitisio` if the `CONTINUE` convention is properly implemented (unless we write something to deal with this specific corruption).

**Comment by esheldon** [comment]
closed via #217
